lib/app/app.rb

```diff
@@ -54,6 +54,8 @@ class ConnectorsWebApp < Sinatra::Base
   end
 
   before do
+    @connector = settings.connector_class.new
```

Comment:
I think with the change you made in `http_call_wrapper` (making `extractor` a method variable instead of instance variable), I think we can revert the changes here? Setting an instance of `http_call_wrapper` to `settings.connector` can avoid the instantiation in each request.

Reply:
Yah, I ended up solving the problem twice, once in the _HttpCallWrapper_ implementation to make it thread-safe, and once in this controller code to make it instantiate a new _HttpCallWrapper_ on every request. I'd prefer to continue instantiating a new object on every request. The shared object approach was surprising and I would expect we'll accidentally introduce new regressions again in the future with it.